### PR TITLE
Conflict Solver Enhancement: Retract conflicting read-only value

### LINKED ISSUE #15610: Retract possible for read-only
Christoph has already provided a POC in PR https://github.com/SAP/spartacus/pull/15512
This has to be enhanced:
1. Retract for this new case should be active also if the general "retract" setting is de-activated.

2. it should consider retractBlocked flag (see CXIEP-754).
This has to be implemented downward compatible (retractBlocked flag is only available as of 22.05).

Tests have not been provided for the enhanced so far in the PR, so tests have to be added.

There is a CPS product to test this, see: https://sap-cx.slack.com/archives/G01KTLWNJ06/p1649140134132159

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts [hunk 1]
```diff
@@ -975,42 +998,66 @@ describe('OccConfiguratorVariantNormalizer', () => {
...
+      sourceAttribute.retractBlocked = true;
+      occConfiguratorVariantNormalizer['addRetractValue'](
+        sourceAttribute,
+        values
+      );
+      expect(values.length).toEqual(0);
+    });
+
+    it("should not add a retract value to the list of values because attribute type is 'NOT_IMPLEMENTED'", () => {
+      configUISettingsConfig.productConfigurator.addRetractOption = true;
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts]: Class has 9 strict mode issues. Please fix.

One way to fix is 

   `   (configUISettingsConfig.productConfigurator ??= {}).addRetractOption =  false;`

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.ts [hunk 2]
```diff
@@ -187,28 +189,60 @@ export class OccConfiguratorVariantNormalizer
...
+        this.uiSettingsConfig?.productConfigurator?.addRetractOption ||
+        (this.isSourceAttributeTypeReadOnly(sourceAttribute) && isConflicting)
       ) {
-        const value: Configurator.Value = {
-          valueCode: OccConfiguratorVariantNormalizer.RETRACT_VALUE_CODE,
-          valueDisplay: '',
-          selected: this.isRetractValueSelected(sourceAttribute),
-        };
+        const attributeType = this.convertAttributeType(
+          sourceAttribute.type ?? OccConfigurator.UiType.NOT_IMPLEMENTED,
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.ts]: Branch not covered with test

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.ts [hunk 3]
```diff
@@ -259,7 +293,11 @@ export class OccConfiguratorVariantNormalizer
     images.push(image);
   }
 
-  convertAttributeType(type: OccConfigurator.UiType): Configurator.UiType {
+  convertAttributeType(
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.ts:286]: Could we extend the signature only by OccConfigurator.Attribute? Then we could later on add more variables that influence the uiType without doing breaking changes

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts (hunk 1 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts]: Done!

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.ts (hunk 2 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.ts]: Done!

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.ts (hunk 3 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.ts:286]: Done!

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.ts [hunk 4]
```diff
@@ -187,28 +185,55 @@ export class OccConfiguratorVariantNormalizer
...
-          selected: this.isRetractValueSelected(sourceAttribute),
-        };
+        const attributeType = this.convertAttributeType(sourceAttribute);
+        if (
+          attributeType === Configurator.UiType.RADIOBUTTON ||
+          attributeType === Configurator.UiType.DROPDOWN
+        ) {
+          const value: Configurator.Value = {
+            valueCode: OccConfiguratorVariantNormalizer.RETRACT_VALUE_CODE,
+            valueDisplay: '',
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.ts]: This line can be removed, right?

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.ts (hunk 4 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.ts]: Done!

**Review by ChristophHi — APPROVED** [review]

**Pull request merged**